uaa/client_test.go

```diff
@@ -260,9 +262,52 @@ var _ = Describe("UAA", func() {
                                                 "name": "some-name",
                                                 "redirect_uri": [ "https://placeholder.example.com" ]
                                         }`), "Expected request body mismatch")
-                                })
-                        })
-			
+					})
+				})
+			})
+
+			When("the definition has allowpulic: true", func() {
+				BeforeEach(func() {
+					uaaConfig.ClientDefinition.AllowPublic = true
+					uaaClient, _ = uaa.New(uaaConfig, trustedCert, skipTLSValidation)
+					createJsonResponse := `{
+				  "scope": [ "admin", "read", "write", "extra-scope" ],
+				  "client_id": "some-client-id",
+				  "resource_ids": ["resource1", "resource2", "some-extra-resource"],
+				  "authorized_grant_types": [ "client_credentials", "password", "token" ],
+				  "authorities": [ "some-authority", "another-authority", "some-extra-authority" ],
+				  "name": "some-name",
+				  "lastModified": 1588809891186,
+				  "required_user_groups": [ ],
+					"allowpublic": true
+				}`
+					createHandler.RespondsWith(http.StatusCreated, createJsonResponse)
+				})
+
+				It("sets allowpublic on the client, and configures a client_secret", func() {
+					actualClient, err := uaaClient.CreateClient("some-client-id", "some-name", "some-space-guid")
+					Expect(err).NotTo(HaveOccurred())
+					Expect(actualClient["allowpublic"]).To(Equal("true"))
+					Expect(actualClient["client_secret"]).To(Equal("-"))
```

Comment:
Hi Ryan, 

Agree that the test description is misleading, have updated it here https://github.com/pivotal-cf/on-demand-service-broker/commit/adbd22698b0891d04568be25f462a0cfb5ac2eb7

The intention here is to satisfy the current (bugged) behaviour in UAA. 

Even for public clients (clients with allowpublic = true), UAA requires that a client_secret is provided when an access token is refreshed. 

Since by definition public clients do not have a client_secret, we set the client_secret value to a bogus value, to satisfy the bug in UAA when refreshing a token

See https://www.pivotaltracker.com/n/projects/2482247/stories/183763390 for more details (log in with google option + your vmware address).

Reply:
Makes sense.  Thanks!